src/tpm.rs

```diff
@@ -196,144 +206,82 @@ Following are function from tpm_quote.py program
+        format!(
+            "tpm2_deluxequote -C {} -L {}:{} -q {} -m {} -s {} -p {} -G {} -P {}",
+            hex::encode(key_handle),
+            hash_alg,
+            pcr_list,
+            hex::encode(nonce),
+            quote_path,
+            sign_path,
+            pcr_path,
+            hash_alg,
+            aik_pw,
+        ),
+        Some(paths)
+    )?;
+
+    let mut quote_list = Vec::new();
+    for val in quotes.values() {
+        quote_list.push(base64_zlib_encode(val.to_string())?);
+    }
+
+    Ok(quote_list.as_slice().join(":"))
 }
 
 /*
- * Input:
- *     nonce string
- *     data that needs to be pass to the pcr
- *     pcrmask
- *
- * Output:
- *     deep quote string from tpm pcr
- *     KeylimeTpmError
- *
- * Getting deep quote form tpm, same implementation as the original python
- * version. Same  procedures as quote by this is a deep quote.
+ * Deep quote is in progress
  */
 pub fn create_deep_quote(
     nonce: String,
     data: String,
     mut pcrmask: String,
     mut vpcrmask: String,
 ) -> Result<String, KeylimeTpmError> {
-    let temp_file = NamedTempFile::new()?;
-    let quote_path = match temp_file.path().to_str() {
-        None => {
-            return Err(KeylimeTpmError::new_tpm_rust_error(
-                "Can't retieve temp file path.",
-            ));
-        }
-        Some(p) => p,
-    };
-    let key_handle = get_tpm_metadata_content("aik_handle")?;
-    let aik_password = get_tpm_metadata_content("aik_pw")?;
-    let owner_password = get_tpm_metadata_content("owner_pw")?;
-
-    if pcrmask == "".to_string() {
-        pcrmask = EMPTYMASK.to_string();
-    }
-
-    if vpcrmask == "".to_string() {
-        vpcrmask = EMPTYMASK.to_string();
-    }
-
-    if !(data == "".to_string()) {
-        let vpcrmask_int: i32 = vpcrmask.parse()?;
-        vpcrmask =
-            format!("0x{}", (vpcrmask_int + (1 << common::TPM_DATA_PCR)));
-        let mut command = format!("pcrreset -ix {}", common::TPM_DATA_PCR);
-
-        //RUN
-        run(command, None)?;
-        let mut hasher = sha::Sha1::new();
-        hasher.update(data.as_bytes());
-        let data_sha1_hash = hasher.finish();
-
-        command = format!(
-            "extend -ix {} -ic {}",
-            common::TPM_DATA_PCR,
-            hex::encode(data_sha1_hash),
-        );
-
-        //RUN
-        run(command, None)?;
-    }
-
-    // store quote into the temp file that will be extracted later
-    let command = format!(
-        "deepquote -vk {} -hm {} -vm {} -nonce {} -pwdo {} -pwdk {} -oq {}",
-        key_handle,
-        pcrmask,
-        vpcrmask,
-        nonce,
-        owner_password,
-        aik_password,
-        quote_path,
-    );
-
-    // RUN
-    let (_, quote_raw) = run(command, Some(quote_path))?;
-    let mut quote_return = String::from("d");
-    quote_return.push_str(&quote_raw);
-    Ok(quote_return)
+    Err(KeylimeTpmError::new_tpm_rust_error(
+        "Deep quote in progress.",
```

Comment:
This looks good so far.  As discussed in person, please update this PR once you have `create_deep_quote` working.

Reply:
Will you merge this one first or until all the TPM2.0 functions are implemented?